apps/ensadmin/src/lib/env.ts

```diff
@@ -1,12 +1,16 @@
 /**
  * Get ENSAdmin service public URL.
+ *
+ * Note: using Vercel platform for ENSAdmin deployments works best when
+ * this function returns undefined and lets default values to be applied.
+ * Read more: https://nextjs.org/docs/app/api-reference/functions/generate-metadata#default-value
  */
-export function ensAdminPublicUrl() {
+export function ensAdminPublicUrl(): string | undefined {
   const envVarName = "ENSADMIN_PUBLIC_URL";
-  const envVarValue = process.env[envVarName];
+  const envVarValue = process.env.ENSADMIN_PUBLIC_URL;
 
   if (!envVarValue) {
-    throw new Error(`Required "${envVarName}" value was not set`);
+    return undefined;
```

Comment:
Making it an optional value to allow Vercel applying its env vars automatically:
https://nextjs.org/docs/app/api-reference/functions/generate-metadata#default-value